frontend/front/src/pages/Public/Components/LangPrefDropdown.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from "react";
+import { useLocation } from "react-router-dom";
+import { Box, Menu, MenuItem, Fade, Button, Typography } from "@mui/material";
+import LanguageIcon from "@mui/icons-material/Language";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import { useTranslation } from "react-i18next";
+
+const LangPrefDropdown = () => {
+  const {
+    t,
+    i18n: { changeLanguage, language },
+  } = useTranslation();
+
+  const [anchorMore, setAnchorMore] = useState(null);
+  const [langDisplay, setLangDisplay] = useState("English");
+  const [currentLanguage, setCurrentLanguage] = useState(language);
+
+  const location = useLocation();
+
+  const langMap = {
+    "en-us": `🇺🇸 ${t("public.global-labels.locales.english")}`,
+    "ht-cr": `🇭🇹 ${t("public.global-labels.locales.creole")}`,
+    "pt-br": `🇧🇷 ${t("public.global-labels.locales.portuguese")}`,
+    "es-us": `🇪🇸 ${t("public.global-labels.locales.spanish")}`,
+  };
```

Comment:
The language selector should not translate language names. Language name should always be in its own language so that speakers of the language can find it (eg. English should always say English, Portuguese should always say Portugues)
![image](https://github.com/codeforboston/urban-league-heat-pump-accelerator/assets/93659375/88f41951-d9cd-41d1-bd81-f9d670060f33)

Here's the Facebook one for reference:
![image](https://github.com/codeforboston/urban-league-heat-pump-accelerator/assets/93659375/62be13db-b091-4142-97b3-48b8aef09d74)

Reply:
I think you're right, but I also feel like all of the text on the screen should be handled in an identical way, even if the values for the text happen to be the same across all languages.

Reply:
Yes, I agree, and I don't think I am translating names. For example, [Massachusetts Clean Energy Center (MassCEC)](https://www.masscec.com/) and Boston Heat Pump—these names remain consistent in all languages. I will double-check to ensure that I'm not translating any names.

Reply:
I think the goal here is to always have the languages written in their own language while still using the translations. You could set all four language files to use this for public.global-labels.locales:
```
"english": "English (United States)",
"portuguese": Português (Brasil)",
"spanish": "Español (Latinoamérica)",
"creole": "Kreyòl (Ayiti)"
```

Reply:
Fixed it: https://github.com/codeforboston/urban-league-heat-pump-accelerator/pull/489/commits/2a089cd364d1b502c4ae3c522afa5393be429f5e